File: backend/app/domain/entities/query_schema.py
from dataclasses import dataclass
from datetime import datetime


@dataclass(slots=True)
class QuerySchemaColumn:
    name: str
    data_type: str
    nullable: bool
    default: str | None = None


@dataclass(slots=True)
class QuerySchemaConstraint:
    column: str
    foreign_table_schema: str
    foreign_table_name: str
    foreign_column_name: str


@dataclass(slots=True)
class QuerySchemaTable:
    name: str
    columns: list[QuerySchemaColumn]
    constraints: list[QuerySchemaConstraint]


@dataclass(slots=True)
class QuerySchema:
    id: str
    name: str
    description: str
    business_rules: str
    context: str
    created_at: datetime
    refreshed_at: datetime
    table_count: int = 0
    column_count: int = 0
    tables: list[QuerySchemaTable] | None = None

File: frontend/src/app/domain/models/query-schema.ts
export interface QuerySchema {
  readonly id: string;
  readonly name: string;
  readonly description: string;
  readonly business_rules: string;
  readonly created_at: string;
  readonly refreshed_at: string;
  readonly table_count: number;
  readonly column_count: number;
}

export interface QuerySchemaColumn {
  readonly name: string;
  readonly data_type: string;
  readonly nullable: boolean;
  readonly default?: string | null;
}

export interface QuerySchemaConstraint {
  readonly column: string;
  readonly foreign_table_schema: string;
  readonly foreign_table_name: string;
  readonly foreign_column_name: string;
}

export interface QuerySchemaTable {
  readonly name: string;
  readonly columns: QuerySchemaColumn[];
  readonly constraints: QuerySchemaConstraint[];
}

export interface QuerySchemaDetail extends QuerySchema {
  readonly tables: QuerySchemaTable[];
}

export interface ImportQuerySchemaPayload {
  readonly name: string;
  readonly description: string;
  readonly businessRules: string;
  readonly file: File;
}
